packages/cli-e2e/__tests__/vue.specs.ts

```diff
@@ -0,0 +1,111 @@
+import retry from 'async-retry';
+import {spawn, spawnSync} from 'child_process';
+
+import type {ChildProcessWithoutNullStreams} from 'child_process';
+
+import {CLI_EXEC_PATH, killCliProcessFamily} from '../utils/cli';
+import {closeAllPages, getBrowser} from '../utils/browser';
+import {Browser, HTTPRequest, Page} from 'puppeteer-core';
+import {loginWithOffice} from '../utils/login';
+import {isSearchRequest} from '../utils/platform';
+
+describe('ui', () => {
+  describe('create:vue', () => {
+    let browser: Browser;
+    const cliProcesses: ChildProcessWithoutNullStreams[] = [];
+    // TODO: CDX-90: Assign a dynamic port for the search token server on all ui projects
+    const clientPort = '8080';
+    const projectName = 'vue-project';
+    const searchPageEndpoint = `http://localhost:${clientPort}`;
+    const tokenProxyEndpoint = `http://localhost:${clientPort}/token`;
+    let interceptedRequests: HTTPRequest[] = [];
+    let page: Page;
+    let waitForProjectToStartTimeout: NodeJS.Timeout;
+
+    beforeAll(async () => {
+      browser = await getBrowser();
+
+      await loginWithOffice(browser, cliProcesses);
+
+      const buildProcess = spawnSync(CLI_EXEC_PATH, [
+        'ui:create:vue',
+        projectName,
+      ]);
+
+      expect(buildProcess.status).toEqual(0);
+
+      const waitForProjectToStart = new Promise<void>((resolve) => {
+        const startServerProcess = spawn('npm', ['run', 'start'], {
+          cwd: projectName,
+          detached: true,
+        });
+
+        cliProcesses.push(startServerProcess);
+
+        waitForProjectToStartTimeout = setTimeout(() => {
+          resolve();
+        }, 15e3);
+      });
+
+      return waitForProjectToStart;
+    }, 240e3);
+
+    beforeEach(async () => {
+      page = await browser.newPage();
+
+      page.on('request', (request: HTTPRequest) => {
+        interceptedRequests.push(request);
+      });
+    });
+
+    afterAll(async () => {
```

Comment:
Similar comment as above about this `afterAll` being potentially generic for all UI commands.